src/core/src/package_managers/AzL3TdnfPackageManager.py

```diff
@@ -105,122 +85,41 @@ def __generate_command_with_snapshotposixtime_if_specified(command_template, sna
-            self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.PACKAGE_MANAGER_FAILURE)
-            if raise_on_exception:
-                raise Exception(error_msg, "[{0}]".format(Constants.ERROR_ADDED_TO_STATUS))
-
-        return out, code
-
     # region Classification-based (incl. All) update check
     def get_all_updates(self, cached=False):
         """Get all missing updates"""
-        self.composite_logger.log_verbose("[TDNF] Discovering all packages...")
+        self.composite_logger.log_verbose("[AzL3TDNF] Discovering all packages...")
         if cached and not len(self.all_updates_cached) == 0:
-            self.composite_logger.log_debug("[TDNF] Get all updates : [Cached={0}][PackagesCount={1}]]".format(str(cached), len(self.all_updates_cached)))
+            self.composite_logger.log_debug("[AzL3TDNF] Get all updates : [Cached={0}][PackagesCount={1}]]".format(str(cached), len(self.all_updates_cached)))
             return self.all_updates_cached, self.all_update_versions_cached  # allows for high performance reuse in areas of the code explicitly aware of the cache
 
         out = self.invoke_package_manager(self.__generate_command_with_snapshotposixtime_if_specified(self.tdnf_check, self.max_patch_publish_date))
         self.all_updates_cached, self.all_update_versions_cached = self.extract_packages_and_versions(out)
-        self.composite_logger.log_debug("[TDNF] Get all updates : [Cached={0}][PackagesCount={1}]]".format(str(False), len(self.all_updates_cached)))
+        self.composite_logger.log_debug("[AzL3TDNF] Get all updates : [Cached={0}][PackagesCount={1}]]".format(str(False), len(self.all_updates_cached)))
         return self.all_updates_cached, self.all_update_versions_cached
 
     def get_security_updates(self):
-        """Get missing security updates. NOTE: Classification based categorization of patches is not available in TDNF as of now"""
-        self.composite_logger.log_verbose("[TDNF] Discovering all packages as 'security' packages, since TDNF does not support package classification...")
+        """Get missing security updates. NOTE: Classification based categorization of patches is not available in Azure Linux as of now"""
+        self.composite_logger.log_verbose("[AzL3TDNF] Discovering all packages as 'security' packages, since TDNF does not support package classification...")
         security_packages, security_package_versions = self.get_all_updates(cached=False)
-        self.composite_logger.log_debug("[TDNF] Discovered 'security' packages. [Count={0}]".format(len(security_packages)))
+        self.composite_logger.log_debug("[AzL3TDNF] Discovered 'security' packages. [Count={0}]".format(len(security_packages)))
         return security_packages, security_package_versions
 
     def get_other_updates(self):
```

Comment:
Generic now - Tdnf

In a hypothetical future where the two behaviors split, that'll have to be addressed based on the realities and specifics of that future. The basic concept is that UTs for AzL3 (today) must prevent future regression in a TDNF only change (by a hypothetical future author + reviewer combo that may not include either of us - i.e. UT is the safety, not tribal knowledge).

Reply:
Still valid in the context of the reply to that comment -- https://github.com/Azure/LinuxPatchExtension/pull/331/files#r2392873773